src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -301,15 +306,15 @@ export function loadOperation(
         const type = body.kind === "Model" ? body : body.type;
         const inputType: InputType = getInputType(
             context,
-            getFormattedType(program, body),
+            getFormattedType(program, type),
```

Comment:
Previously `type` is never used. I think here passing in `body` is wrong. For primitive types, it should be `body.type` which is `type`. Changing to `type` fixes an issue, see the test case `encoding/bytes` below.

Reply:
No, we should not change it to `type`, because we need to get the encoding information, encoding information could decorated on Model or ModelProperty.

Reply:
roll back changes per discussion